pkg/apis/networkoperator/v1/types.go

```diff
@@ -119,6 +123,59 @@ type OpenshiftSDNConfig struct {
 	UseExternalOpenvswitch *bool `json:"useExternalOpenvswitch,omitempty"`
 }
 
+type OpenStackCredentials struct {
+	// TODO(dulek): Might want to add all the options, e.g. keys.
+	AuthURL           string `json:"authURL"`
+	UserDomainName    string `json:"userDomainName"`
+	Username          string `json:"username"`
+	Password          string `json:"password"`
+	ProjectDomainName string `json:"projectDomainName"`
+	ProjectId         string `json:"projectId"`
+}
+
+// KuryrConfig configures the Kuryr-Kubernetes SDN
+type KuryrConfig struct {
+	// The base64-encoded CA certificate to use when connecting to OpenStack.
+	// +optional
+	OpenStackCACertificate string `json:"openStackCACertificate"`
+
+	OpenStackCredentials *OpenStackCredentials `json:"openStackCredentials"`
+
+	// If true, kuryr-daemon readiness and liveness probes are enabled.
+	// +optional
+	DaemonEnableProbes *bool `json:"daemonEnableProbes"`
+
+	// The port kuryr-daemon will listen for readiness and liveness requests.
+	// +optional
+	DaemonProbesPort *uint16 `json:"daemonProbesPort"`
+
+	// The CNI bin directory on the hosts, usually /opt/cni/bin.
+	CNIBinDir string `json:"cniBinDir"`
```

Comment:
Users shouldn't be allowed to configure this; we should treat it as a constant we own.

Reply:
Okay, great.